sdk/tests/core/test_az_credential_client_secret.c

```diff
@@ -121,9 +143,21 @@ az_result send_request(_az_http_request const* request, az_http_response* respon
           AZ_SPAN_FROM_STR("https://www.microsoft.com/test/request"),
           request_url)) // Auth request
   {
-    assert_true(az_span_is_content_equal(
-        AZ_SPAN_FROM_STR("https://login.microsoftonline.com/TenantID/oauth2/v2.0/token"),
-        request_url));
+    {
+      uint8_t auth_url_buf[200] = { 0 };
+      az_span az_auth_url = AZ_SPAN_FROM_BUFFER(auth_url_buf);
+
+      {
+        az_span auth_url_remainder = az_span_copy(az_auth_url, authority);
+
+        auth_url_remainder
+            = az_span_copy(auth_url_remainder, AZ_SPAN_FROM_STR("TenantID/oauth2/v2.0/token"));
+
+        az_auth_url = az_span_slice(
+            az_auth_url, 0, (int32_t)(az_span_ptr(auth_url_remainder) - az_span_ptr(az_auth_url)));
```

Comment:
nit: Correct me if I am mistaken, but I think this should use `az_span_size` rather than `az_span_ptr`.
```suggestion
        az_auth_url = az_span_slice(
            az_auth_url, 0, az_span_size(az_auth_url) - az_span_size(auth_url_remainder));
```